contracts/RiskSteward/MarketCapsRiskSteward.sol

```diff
@@ -0,0 +1,198 @@
+    event SupplyCapUpdated(address market, uint256 newSupplyCap);
+
+    /**
+     * @notice Emitted when a borrow cap is updated
+     */
+    event BorrowCapUpdated(address market, uint256 newBorrowCap);
+
+    /**
+     * @notice Emitted when the max increase bps is updated
+     */
+    event MaxIncreaseBpsUpdated(uint256 oldMaxIncreaseBps, uint256 newMaxIncreaseBps);
+
+    /**
+     * @notice Thrown when a maxIncreaseBps value of 0 is set
+     */
+    error InvalidMaxIncreaseBps();
+
+    /**
+     * @notice Thrown when an updateType that is not supported is operated on
+     */
+    error UnsupportedUpdateType();
+
+    /**
+     * @notice Thrown when the new value of an update is our of range
+     */
+    error UpdateNotInRange();
+
+    /**
+     * @notice Thrown when the update is not coming from the RiskStewardReceiver
+     */
+    error OnlyRiskStewardReceiver();
+
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor(address corePoolComptroller_, address riskStewardReceiver_) {
+        require(corePoolComptroller_ != address(0), "Core Pool Comptroller address must not be zero");
+        CORE_POOL_COMPTROLLER = ICorePoolComptroller(corePoolComptroller_);
+        RISK_STEWARD_RECEIVER = IRiskStewardReceiver(riskStewardReceiver_);
+        _disableInitializers();
+    }
+
+    function initialize(address accessControlManager_, uint256 maxIncreaseBps_) external initializer {
+        __Ownable2Step_init();
+        __AccessControlled_init_unchained(accessControlManager_);
+        if (maxIncreaseBps_ == 0) {
+            revert InvalidMaxIncreaseBps();
+        }
+        maxIncreaseBps = maxIncreaseBps_;
+    }
+
+    function setMaxIncreaseBps(uint256 maxIncreaseBps_) external {
+        _checkAccessAllowed("setMaxIncreaseBps(uint256)");
+        uint256 _maxIncreaseBps = maxIncreaseBps;
+        if (maxIncreaseBps_ == 0) {
+            revert InvalidMaxIncreaseBps();
+        }
+        maxIncreaseBps = maxIncreaseBps_;
+        emit MaxIncreaseBpsUpdated(_maxIncreaseBps, maxIncreaseBps_);
+    }
+
+    /**
+     * @notice
+     * @param update
+     */
+    function processUpdate(RiskParameterUpdate calldata update) external {
+        if (msg.sender != address(RISK_STEWARD_RECEIVER)) {
+            revert OnlyRiskStewardReceiver();
+        }
+        if (Strings.equal(update.updateType, "supplyCap")) {
+            _processSupplyCapUpdate(update);
+        } else if (Strings.equal(update.updateType, "borrowCap")) {
+            _processBorrowCapUpdate(update);
+        } else {
+            revert UnsupportedUpdateType();
+        }
+    }
+
+    function _updateSupplyCaps(address market, bytes memory newValue) internal {
+        address comptroller = IVToken(market).comptroller();
+        address[] memory newSupplyCapMarkets = new address[](1);
+        newSupplyCapMarkets[0] = market;
+        uint256[] memory newSupplyCaps = new uint256[](1);
+
+        newSupplyCaps[0] = uint256(bytes32(newValue));
```

Comment:
I think we are decoding the `newValue` in a different way here and in `_validateSupplyCapUpdate`. Any reason?

Reply:
I updated this but it must not have been pushed